src/main/java/org/swisspush/redisques/util/QueueStatisticsCollector.java

```diff
@@ -45,28 +49,79 @@
     private final Map<String, AtomicLong> queueFailureCount = new HashMap<>();
     private final Map<String, Long> queueBackpressureTime = new HashMap<>();
     private final Map<String, Long> queueSlowDownTime = new HashMap<>();
-
+    private final ConcurrentMap<String, AtomicLong> queueMessagePaceCtr = new ConcurrentHashMap<>();
+    private volatile Map<String, Long> queueMessagePace = new HashMap<>();
     private final RedisAPI redisAPI;
     private final LuaScriptManager luaScriptManager;
     private final String queuePrefix;
+    private final Vertx vertx;
 
     public QueueStatisticsCollector(RedisAPI redisAPI, LuaScriptManager luaScriptManager,
-            String queuePrefix) {
+            String queuePrefix, Vertx vertx, int paceIntervalTime) {
         this.redisAPI = redisAPI;
         this.luaScriptManager = luaScriptManager;
         this.queuePrefix = queuePrefix;
+        this.vertx = vertx;
+        paceStatisticsScheduler(paceIntervalTime);
     }
 
     /**
-     * Does reset all statistics values of the given queue. In memory but as well the persisted
+     * This scheduled task collects and calculates the message pace per queue in the local system
+     * within the past time interval.
+     *
+     * NOTE: The pace values are held in memory only due to performance reasons. We don't want to
+     * store regularly a lot of values (all queues) for pace statistics only.
+     * This would no more work properly of course if there are multiple Redisques instances deployed.
+     * In such case, the client interested in such values must query each instance for itself.
+     *
+     * By default, the pace interval is configured to 1 Minute currently.
+     */
+    private void paceStatisticsScheduler(int paceIntervalTime){
```

Comment:
maybe name the argument paceIntervalTimeMinutes

Reply:
the unit for the `setPeriodic` method is milliseconds, not minutes. Either change the name of the parameter to `paceIntervalMs` or calculate to milliseconds before use.

Reply:
Will rename all over with Ms to make it clear everywhere.